modules/generic/meta-maintainers.nix

```
@@ -0,0 +1,61 @@
+# Test:
+#   ./meta-maintainers/test.nix
+{ lib, ... }:
+let
+  inherit (lib)
+    mkOption
+    mkOptionType
+    types
+    ;
+
+  maintainer = mkOptionType {
+    name = "maintainer";
+    check = email: lib.elem email (lib.attrValues lib.maintainers);
+    merge =
+      loc: defs:
+      lib.listToAttrs (lib.singleton (lib.nameValuePair (lib.last defs).file (lib.last defs).value));
```

Style nit: this feels unnecessarily complex just to get it formatted on a single line?

Or at least, using `listToAttrs` with an explicit singleton feels like a code smell.

```suggestion
      {
        ${(lib.last defs).file} = (lib.last defs).value;
      }
```

I also feel like the merge function only looking at the `last` definition requires some explanation and/or justification, perhaps as a comment?

---

Looking closer, I see this is copied as-is from the existing impl. So this isn't really feedback on your PR, but on the existing code 😅

---

I've done that countless times, but thanks, it works!